File: DemoProject/DemoPreoject/CodeFeed/EntityCode/ModuleAcct/acct/C/BaseAcctViewController.m
//
//  BaseAcctViewController.m
//  ExampleProjectDemo
//
//  Created by BlanBok on 2018/1/15.
//  Copyright © 2018年 juyuanGroup. All rights reserved.
//

#import "BaseAcctViewController.h"
#import "BaseAcctDataModelIndex.h"
#import "BaseAcctViewIndex.h"
#import "BaseAcctViewModelIndex.h"

@interface BaseAcctViewController ()

@property (nonnull,nonatomic,strong) BaseAcctDataModelIndex *index_dataModel;

@property (nonnull,nonatomic,strong) BaseAcctViewIndex *index_view;

@property (nonnull,nonatomic,strong) BaseAcctViewModelIndex *index_viewModel;


@end

@implementation BaseAcctViewController


- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = CUSTOM_COLOR_HEX(0xf4f4f4);
    [self init_commit];
}

- (void)init_commit
{
    [self setIndex_dataModel];
    [self setIndex_view];
    [self setIndex_viewModel];
}

#pragma mark-init
- (void)setIndex_dataModel
{
    
}

- (void)setIndex_view
{
    self.index_view.container = self.view;
}

- (void)setIndex_viewModel
{
    
}

#pragma mark-lazyload
- (BaseAcctDataModelIndex *)index_dataModel
{
    if (_index_dataModel == nil) {
        _index_dataModel = [BaseAcctDataModelIndex new];
    }
    return _index_dataModel;
}

- (BaseAcctViewIndex *)index_view
{
    if (_index_view == nil) {
        _index_view = [BaseAcctViewIndex new];
    }
    return _index_view;
}

- (BaseAcctViewModelIndex *)index_viewModel
{
    if (_index_viewModel == nil) {
        _index_viewModel = [BaseAcctViewModelIndex new];
    }
    return _index_viewModel;
}


@end

File: DemoPreoject/DemoPreoject/CodeFeed/EntityCode/ModuleAcct/Acct/C/BaseAcctViewController.swift
//
//  BaseAcctViewController.swift
//  OrderTreasure
//
//  Created by ZhangJian on 2018/4/20.
//  Copyright © 2018年 ZhangJian. All rights reserved.
//
// 个人信息页

import UIKit

class BaseAcctViewController: UIViewController {

//    private(set) lazy var index_DataModel    = BaseAcctDataModelIndex()
    
    private(set) lazy var index_view         = BaseAcctViewIndex()
    
//    private(set) lazy var index_viewModel    = BaseAcctViewModelIndex()
    
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.view.backgroundColor = RGBA(246, 246, 246, 1)
        self.commit_init()
        
        // Do any additional setup after loading the view.
    }
    
    func commit_init() {
        self.configure_viewIndex()
        self.configure_viewModelIndex()
    }
    
    func configure_viewIndex() {
        self.index_view.setContainerView(self.view)
        
    }
    
    func configure_viewModelIndex() {
        
    }
        
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.setNavigationBarHidden(true, animated: animated)
    }
    


}
